src/components/configMaps/ConfigMap.tsx

```diff
@@ -233,35 +236,61 @@ export function ConfigMapForm({ id, appId, name = "", external, data = null, typ
     async function handleDelete() {
         try {
             if (!envId) {
-                const { result } = await deleteConfig(id, appId, name)
+                await deleteConfig(id, appId, name)
                 toast.success('Successfully deleted')
                 updateForm(listIndex, null)
             }
             else {
-                const { result } = await deleteEnvironmentConfig(id, appId, envId, name);
+                await deleteEnvironmentConfig(id, appId, envId, name);
                 toast.success('Successfully deleted')
                 updateForm(true)
             }
-
         }
         catch (err) {
             showError(err)
         }
     }
 
+    function handleFilePermission(e): void {
+        let permissionValue = e.target.value;
+        setFilePermissionValue({ value: permissionValue, error: "" });
+    }
+
     async function handleSubmit(e) {
+        const configmapNameRegex = new RegExp(/^[-.a-zA-Z0-9]+$/);
+
         if (!configName.value) {
             setName({ value: "", error: 'Field is manadatory' })
             return
         }
-        if (!/^[-.a-zA-Z0-9]+$/.test(configName.value)) {
+        if (!configmapNameRegex.test(configName.value)) {
             setName({ value: configName.value, error: 'Name must be of format /^[-.a-zA-Z0-9]+$/' })
```

Comment:
we should rather put statement, like configmap name should start with lower case alphabet and can contain numbers, alphabets, - and .